src/consensus/pbft_manager.cpp

```diff
@@ -1442,8 +1442,6 @@ void PbftManager::pushSyncedPbftBlocksIntoChain_() {
         break;
       }
 
-      syncBlockQueuePop();
```

Comment:
This is not good as you just introduced the race condition by this change. There is relatively long time between `processSyncBlock()` and `pushPbftBlock_` when the the sync block is neither in sync_queue nor in database and there is some code that is checking current sync pbft period. This code wont work properly now... It was popped intentionally after  pushPbftBlock_